shell/platform/android/io/flutter/embedding/engine/plugins/shim/ShimRegistrar.java

```diff
@@ -165,6 +165,7 @@ public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     }
 
     pluginBinding = null;
+    activityPluginBinding = null;
```

Comment:
FYI, this isn't technically necessary because the engine will already be detached from the Activity before this executes.

Reply:
Discussed offline. Understood. 

Though I'd keep them here since the current docs don't refer have the engine and activity callbacks refer to each other in terms of timing. So it would be a knowledge leak from the implementer side here to assume there is a sequence. 

(though not a bad idea to actually add that to the javadoc, e.g. onDetachedFromEngine is guaranteed to call after all the ui component bindings had a chance to detach first and vice versa).